packages/@react-aria/selection/src/DOMLayoutDelegate.ts

```diff
@@ -24,7 +24,7 @@ export class DOMLayoutDelegate implements LayoutDelegate {
     if (!container) {
       return null;
     }
-    let item = key != null ? container.querySelector(`[data-key="${CSS.escape(key.toString())}"]`) : null;
+    let item = key != null ? container.querySelector(`[data-key$="${CSS.escape(key.toString())}"]`) : null;
```

Comment:
Not really happy with this, but passing `idScope` down here will be a mess.

Reply:
what if you apply the collection's `idScope` to the collection node as its own `data-key` and look that up via the container ref? That way you can construct the expected node data key here. It might be good to make a util function that generalizes the construction of the nodeKey just so we can make sure we are consistent with the `${idScope}-${manager.focusedKey.toString()}` structure.

Reply:
sure, I'll ask if anyone one remembers any other areas where we might have assumptions for the application of `data-key`. Does feel a bit iffy to pull in the `@react-aria/selection` package into the `@react-aria/dnd` package just for this util, I'll let others weight in there. If you'd like you could skip using that function for the dnd package for now, can be followup